engine/src/cmd/damageable.cpp

```diff
@@ -251,10 +251,8 @@ void Damageable::ApplyDamage( const Vector &pnt,
         static bool player_autoeject =
                 GameConfig::GetVariable( "physics", "player_autoeject", true );
 
-        if(shot_at_is_player && player_autoeject) {
-            unit->EjectCargo( (unsigned int) -1 );
-        } else if(rand() < (RAND_MAX*auto_eject_percent) && unit->isUnit() == _UnitType::unit &&
-                  unit->faction != neutralfac && unit->faction != upgradesfac) {
+        if (shot_at_is_player && player_autoeject && rand() < (RAND_MAX*auto_eject_percent)
```

Comment:
This will only apply to the player. The current behavior is that other units auto eject some of the time.

Reply:
Not sure I follow?

Reply:
The deleted code was split in two:
if(player && player_auto_eject) eject
else something...

Your new code has merged both so the if statement will only apply to the player ship. Other pilots will not eject.

Reply:
Hmm. You're probably right. Let me try this again